osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs

```diff
@@ -50,7 +50,7 @@ public partial class OsuModFlashlight : ModFlashlight<OsuHitObject>, IApplicable
         public void ApplyToDrawableHitObject(DrawableHitObject drawable)
         {
             if (drawable is DrawableSlider s)
-                s.Tracking.ValueChanged += _ => flashlight.OnSliderTrackingChange(s);
+                s.OnUpdate += _ => flashlight.OnSliderTrackingChange(s);
```

Comment:
Since the problem here lays down in `OnSliderTrackingChange` being called before slider start time, how about scheduling the change inside `OnSliderTrackingChange`? e.g.
```diff
diff --git a/osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs b/osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs
index 5a6cc50082..e20cdbbf51 100644
--- a/osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs
+++ b/osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs
@@ -50,7 +50,7 @@ public partial class OsuModFlashlight : ModFlashlight<OsuHitObject>, IApplicable
         public void ApplyToDrawableHitObject(DrawableHitObject drawable)
         {
             if (drawable is DrawableSlider s)
-                s.OnUpdate += _ => flashlight.OnSliderTrackingChange(s);
+                s.Tracking.ValueChanged += _ => flashlight.OnSliderTrackingChange(s);
         }
 
         private partial class OsuFlashlight : Flashlight, IRequireHighFrequencyMousePosition
@@ -69,9 +69,14 @@ public OsuFlashlight(OsuModFlashlight modFlashlight)
             public void OnSliderTrackingChange(DrawableSlider e)
             {
                 // If a slider is in a tracking state, a further dim should be applied to the (remaining) visible portion of the playfield.
-                FlashlightDim = Time.Current >= e.HitObject.StartTime && e.Tracking.Value ? 0.8f : 0.0f;
+                if (Time.Current >= e.HitObject.StartTime)
+                    updateFlashlightDim(e.Tracking.Value);
+                else
+                    Scheduler.AddDelayed(updateFlashlightDim, e.Tracking.Value, e.HitObject.StartTime - Time.Current);
             }
 
+            private void updateFlashlightDim(bool tracking) => FlashlightDim = tracking ? 0.8f : 0.0f;
+
             protected override bool OnMouseMove(MouseMoveEvent e)
             {
                 var position = FlashlightPosition;
```

Reply:
I'd rather not. The current solution may be nuclear but at least it's simple.

Reply:
I'm not sure I agree? Making it run every single update frame for every single slider is way too hacky for me compared to just making the thing run when it's supposed to run? You might also be able to get it working with `BeginAbsoluteSequence` and transform `FlashlightDim` instead of scheduling, but I don't see the point in `OnUpdate` to be honest.

Reply:
Disregarding performance concerns, there's NO WAY a scheduler's `AddDelayed` will work as you expect it to. It won't handle rewind or pause or anything.

Reply:
It will...it's part of `DrawableRuleset`'s frame stability container. It's timed alongside every single gameplay component there.

And regarding performance concerns, this method fires off whenever a slider changes its tracking state. I'm not an expert when it comes to delegate memory footprint or whatever other performance concerns that comes, but if such callback qualifies as bad for scheduling, then sure...`OnUpdate` I guess.

Reply:
I continue to think that while the current solution may be a touch primitive it is still better (in being less prone to failure) than any of the proposed alternatives.

Reply:
I'll take back my proposal of using schedulers because of the potential performance concerns from delegate allocations and what not.

Reply:
> It will...it's part of DrawableRuleset's frame stability container. It's timed alongside every single gameplay component there (i.e. Flashlight.Clock = FrameStableClock).

I have zero faith in this.